pkg/kubelet/kubelet.go

```diff
@@ -1041,6 +1057,64 @@ type Kubelet struct {
 	dockerLegacyService dockershim.DockerLegacyService
 }
 
+func initializeCertificateManager(kubeClient clientset.Interface, kubeCfg *componentconfig.KubeletConfiguration, nodeName types.NodeName, hostname string) (certificate.Manager, error) {
+	var certSigningRequestClient clientcertificates.CertificateSigningRequestInterface
+	if kubeClient != nil && kubeClient.Certificates() != nil {
+		certSigningRequestClient = kubeClient.Certificates().CertificateSigningRequests()
+	}
+	certificateStore, err := certificate.NewFileStore(
+		"kubelet-server",
+		kubeCfg.CertDirectory,
+		kubeCfg.CertDirectory,
+		kubeCfg.TLSCertFile,
+		kubeCfg.TLSPrivateKeyFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize certificate store: %v", err)
+	}
+	ips, err := allUnicastIPs()
+	if err != nil {
+		return nil, err
+	}
+	return certificate.NewManager(&certificate.Config{
+		CertificateSigningRequestClient: certSigningRequestClient,
+		Template: &x509.CertificateRequest{
+			Subject: pkix.Name{
+				CommonName:   string(nodeName),
+				Organization: []string{"system:nodes"},
+			},
+			DNSNames:    []string{hostname},
+			IPAddresses: ips,
```

Comment:
I meant, is that intended? Do we want the loopback address on the certificate? Maybe the answer is yes, so that it can be used over loopback. I'm just not that familiar with the certificates API and usage :)